pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/PicklistQuestionCachedDao.java

```diff
@@ -56,33 +57,37 @@ public TemplateDao getTemplateDao() {
 
     @Override
     public Map<Long, GroupAndOptionDtos> findOrderedGroupAndOptionDtos(Iterable<Long> questionIds, long timestamp) {
-        if (isNullCache(questionIdToGroupAndOptionsCache)) {
+        if (isNullCache(questionVersionKeyToGroupAndOptionsCache)) {
             return delegate.findOrderedGroupAndOptionDtos(questionIds, timestamp);
         } else {
             Set<Long> missingQuestionIds = new HashSet<>();
             Map<Long, GroupAndOptionDtos> result = new HashMap<>();
 
             for (var questionId : questionIds) {
+                String key = questionId + ":" + timestamp;
                 try {
-                    GroupAndOptionDtos cachedDto = questionIdToGroupAndOptionsCache.get(questionId);
+                    GroupAndOptionDtos cachedDto = questionVersionKeyToGroupAndOptionsCache.get(key);
```

Comment:
The fix is to cache using a composite key (made from `questionId` and a `timestamp` which indirectly represents the version) rather than solely relying on `questionId`.